mne/beamformer/_lcmv.py

```diff
@@ -121,52 +120,38 @@ def make_lcmv(info, forward, data_cov, reg=0.05, noise_cov=None, label=None,
            brain imaging (2008) Springer Science & Business Media
     """
     picks = _check_info_inv(info, forward, data_cov, noise_cov)
+    # check number of sensor types present in the data and ensure a noise cov
+    noise_cov = _check_one_ch_type(info, picks, noise_cov, 'lcmv').copy()
+    if 'estimator' in noise_cov:
+        del noise_cov['estimator']
 
     data_rank = compute_rank(data_cov, rank=rank, info=info)
-    if noise_cov is not None:
-        noise_rank = compute_rank(noise_cov, rank=rank, info=info)
-        for key in data_rank:
-            if key not in noise_rank or data_rank[key] != noise_rank[key]:
-                raise ValueError('%s data rank (%s) did not match the noise '
-                                 'rank (%s)'
-                                 % (key, data_rank[key],
-                                    noise_rank.get(key, None)))
-        del noise_rank
+    noise_rank = compute_rank(noise_cov, rank=rank, info=info)
```

Comment:
Is ``noise_cov`` mandatory now? I would vote against that - there is heaps of cases where one does not want to whiten in beamforming and we should keep that possibility.

Reply:
No, `noise_cov` can still be none if and only if there is one channel type.

Internally we just convert this to a diagonal matrix, though, as it removes a number of conditionals but should otherwise behave equivalently. (The projection, for example, gets applied to it, so `compute_whitener` ends up basically just giving you a projection operator.)

Reply:
Okay, that is good. I will do another pass on the code re LCMV later tonight.